bin/improver-standardise

```diff
@@ -0,0 +1,177 @@
+                        help="The output path for the processed NetCDF")
+
+    parser.add_argument("--target_grid_filepath", metavar="TARGET_GRID",
+                        help='If specified then regridding of the source '
+                             'against the target grid is enabled.')
+
+    parser.add_argument("--fix_float64", action='store_true', default=False,
+                        help="Check and fix cube for float64 data. Without "
+                             "this option an exception will be raised if "
+                             "float64 data is found but no fix applied.")
+
+    parser.add_argument("--nearest", action='store_true', default=False,
+                        help='If True, regridding will be performed using '
+                             'iris.analysis.Nearest() instead of Linear(). '
+                             'Use for less continuous fields, '
+                             'e.g. precipitation.')
+
+    parser.add_argument("--extrapolation_mode", default='nanmask',
+                        help='Mode to use for extrapolating data into regions '
+                             'beyond the limits of the source_data domain. '
+                             'Modes are: '
+                             'extrapolate - The extrapolation points will '
+                             'take their value from the nearest source point. '
+                             'nan - The extrapolation points will be be '
+                             'set to NaN. '
+                             'error - A ValueError exception will be raised, '
+                             'notifying an attempt to extrapolate. '
+                             'mask  - The extrapolation points will always be '
+                             'masked, even if the source data is not a '
+                             'MaskedArray. '
+                             'nanmask - If the source data is a MaskedArray '
+                             'the extrapolation points will be masked. '
+                             'Otherwise they will be set to NaN. '
+                             'Defaults to nanmask.')
+
+    parser.add_argument("--json_file", metavar="JSON_FILE", default=None,
+                        help='Filename for the json file containing required '
+                             'changes that will be applied '
+                             'to the metadata. Defaults to None.')
+
+    args = parser.parse_args()
+
+    # source file data path is a mandatory argument
+    source_data = load_cube(args.source_data_filepath)
+
+    if args.fix_float64:
+        check_cube_not_float64(source_data, fix=True)
+    else:
+        check_cube_not_float64(source_data, fix=False)
+
+    if args.target_grid_filepath or args.json_file or args.fix_float64:
+        if not args.output_filepath:
+            msg = ("An argument has been specified that requires an output "
+                   "filepath but none has been provided")
+            raise ValueError(msg)
```

Comment:
Please add a BATS test to trigger this error.

Reply:
The `ValueError` message generated also comes along with a lot of Iris warnings about `Gracefully filling 'latitude' dimension coordinate masked points`. Presumably I would just want the test to check for the existence of `ValueError: An argument has been specified that requires an output filepath but none has been provided` within the response but I'm not sure how to do this with the bats script. 

I've tried using asterix for the test comparison such as `[[ "$output" =~ "*$expected*" ]]`
but it doesn't work.

Reply:
I think that you could copy this example: https://github.com/metoppv/improver/blob/master/tests/improver-orographic-enhancement/05-height_level_not_found.bats. As in this example, only a partial match to the ValueError is being tested, so as long as the `$expected` value e.g. `ValueError: An argument has been specified that requires an output filepath but none has been provided` is somewhere within `$output` then the match should be successful.